src/lightning/fabric/plugins/precision/amp.py

```diff
@@ -50,15 +51,19 @@ def __init__(
         self.device = device
         self.scaler = scaler
 
+        precision_to_type = {"bf16-mixed": torch.bfloat16, "16-mixed": torch.float16}
```

Comment:
Continuing with the microoptimizations, instead of creating a dict and accessing it:

```python
self.dtype = torch.bfloat16 if self.precision == "bf16-mixed" else torch.half
```

Reply:
The current way is more readable IMO, especially when there are more than two choices (like in the deepspeed precision plugin). The plugin is only instantiated once, so this won't speed up training.